Examples/MAX32665/SPIXF_SFCC/main.c

```diff
@@ -0,0 +1,253 @@
+    *ssec_elapsed = (MXC_F_RTC_SSEC_SSEC - start_ssec) + stop_ssec;
+
+    if (*ssec_elapsed > MXC_F_RTC_SSEC_SSEC) {
+        *sec_elapsed += 1;
+        *ssec_elapsed -= MXC_F_RTC_SSEC_SSEC;
+    }
+
+    //Print Results
+    printf("Test %d Complete!\n", test_num);
+    printf("Execution Time: %d.%ds\n\n", *sec_elapsed, SSEC_TO_MSEC(*ssec_elapsed));
+
+    return;
+}
+
+/* ************************************************************************** */
+int main(void)
+{
+    int err;
+    uint32_t id;
+    void (*func)(void);
+    uint32_t test1_sec, test1_ssec;
+    uint32_t test2_sec, test2_ssec;
+
+    printf("\n\n********************* SPIXF/SFCC Example *********************\n");
+    printf("This example demonstrates the performance benefits of enabling the\n");
+    printf("SFCC when executing from the %s external flash chip.\n\n", EXT_FLASH_NAME);
+
+    // Initialize the RTC (used as an execution timer)
+    if (MXC_RTC_Init(0, 0) != E_NO_ERROR) {
+        printf("Failed to initialize RTC.\n");
+        printf("Examples failed.\n");
+        return E_BAD_STATE;
+    }
+    MXC_RTC_Start();
+
+    // Initialize interface (SPIXF) with external flash chip
+    if (Ext_Flash_Init() != E_NO_ERROR) {
+        printf("External flash initialization Failed\n");
+        printf("Example Failed\n");
+        while (1) {}
+    }
+    printf("%s Initialized.\n", EXT_FLASH_NAME);
+    Ext_Flash_Reset();
+
+    // Check ID of the external flash
+    if ((id = Ext_Flash_ID()) != EXT_FLASH_EXP_ID) {
+        printf("Error verifying external flash ID: 0x%x\n", id);
+        printf("Example Failed\n");
+        while (1) {}
+    }
+
+    // Erase sector of external flash to store test function
+    if ((err = Ext_Flash_Erase(0x00000, Ext_Flash_Erase_64K)) != E_NO_ERROR) {
+        printf("Flash erase failed with error code: %d\n", err);
+        printf("Example Failed\n");
+        while (1) {}
+    }
+
+    // Enable Quad mode if necessary
+    if (EXT_FLASH_SPIXFC_WIDTH == Ext_Flash_DataLine_Quad) {
+        if (Ext_Flash_Quad(1) != E_NO_ERROR) {
+            printf("Error enabling quad mode\n\n");
+            while (1) {}
+        }
+    } else {
+        if (Ext_Flash_Quad(0) != E_NO_ERROR) {
+            printf("Error disabling quad mode\n\n");
+            while (1) {}
+        }
+    }
+
+    // Load test function into external flash and set function pointer to it's location
+    printf("Loading test function into external flash.\n\n", (uint32_t)(&__load_length_xip),
+           &__load_start_xip);
+    if ((err = Ext_Flash_Program_Page(EXT_FLASH_ADDR, &__load_start_xip,
+                                      (uint32_t)(&__load_length_xip), EXT_FLASH_SPIXFC_WIDTH)) !=
+        E_NO_ERROR) {
+        printf("Error Programming: %d\n", err);
+        fail++;
+    }
+    func = (void (*)(void))(MXC_XIP_MEM_BASE | 0x1);
+
+    // Setup SPIX
+    spixf_cfg_setup();
+
+    printf("Setup complete. Press SW2 to run SFCC test.\n\n");
```

Comment:
Is SW2 correct for all possible boards?

Reply:
Yes and no. The push button at index 0 is not labeled "SW2" on every EV kit, but this example is only compatible with the MAX32665 at the moment. So it is correct for all possible boards right now, but there could be another chip in the future that is compatible with this example and it does not call PB0 "SW2".

Reply:
Other examples also have a pushbutton name hardcoded in the printf, maybe we should define those in board.h?

Reply:
Sounds good.  No change needed then.  Agreed - we should work out a long-term solution in board.h to associate a string with the pushbuttons.  It's a common enough gotcha to make it worth it.